### DISCUSSION TITLE
HTTPS cert installation for LE

### DISCUSSION THREAD

**Discussion by BustedSec** [Security-Onion-Solutions/securityonion#2883] [Unsupported Versions] [answered]
I am unable to locate in the documentation how to replace the default self-signed certificate. I would like to use certbot and generate a new letsencrypt cert to avoid browser certificate errors. 

Please advice.

**Comment by BustedSec** [comment]
I found some very basic info at https://github.com/Security-Onion-Solutions/securityonion/issues/1766 and updated it with my notes.

**Comment by BustedSec** [comment, marked as answer]
Long story short how I got this to work: 
1. stop nginx (sudo so-nginx-stop) then download certbot-(apt\yum install certbot). 
2. open up your iptables to allow certbot in -and run certbot certonly --standalone. 
3. it generates files in a subfolder of /ect/letsencrypt/live/ with your site name. cd to it. 
3. copy fullchain.pem to /etc/pki/managerssl.crt 
4. copy privkey.pem to /etc/pki/managerssl.key. 
5. Then reboot (to restore original iptables config)

**Reply by facyber** [reply]
Thanks for the tips, this worked for me, now just need to automate this proces.

**Reply by Xboarder56** [reply]
This is what I got so far @facybet. Gonna add in the final command "so-nginx-restart" once I verify it works after replacing the certificate. 

Note: I found an issue with osquery when the cert was replaced, I'm still trying to track this but I think ca.crt/ca.key have to match the LE one as well. 


# Security Onion - Cert - v0.1

CERT="/home/so/securityonion.org/cert.pem"
PRIVKEY="/home/so/securityonion.org/key.pem"
dest_cert_dir="/etc/pki/"

if test -f "$CERT"; then
    echo "Found " $CERT
        cp $CERT $dest_cert_dir"managerssl.crt"
        cp $PRIVKEY $dest_cert_dir"managerssl.key"
    echo "Successfully updated Security Onion Cert"
else
        echo "No cert found. Exiting"
fi

**Reply by facyber** [reply]
Hey @Xboarder56 you managed to get Fleet working? Does replacing ca.crt and ca.key helps? There will be documented way (#3771) to make it working with other certs in the 2.3.50 version, but I was wondering if you already managed to make it work.  :)

**Reply by Xboarder56** [reply]
I ended up having to copy the cert to the os query endpoint every time the security onion console one changed.

**Reply by facyber** [reply]
Same cert.pem file you used for managerssl.crt? What do you mean console one changed?

**Reply by Xboarder56** [reply]
Yeah exactly. I mean that when the 90 days is up on the let’s encrypt certificate on the console all the clients using fleet/osquery have to replace their certificate with the new one.

**Reply by facyber** [reply]
Hah, I did the same thing and there are no more error logs for certificate, but error logs for extension manager are still present, even with self-generated SO certs...

Still thanks for the help, will check it out further.

**Reply by hacker0ni** [reply]
> Long story short how I got this to work:
> 
> 1. stop nginx (sudo so-nginx-stop) then download certbot-(apt\yum install certbot).
> 2. open up your iptables to allow certbot in -and run certbot certonly --standalone.
> 3. it generates files in a subfolder of /ect/letsencrypt/live/ with your site name. cd to it.
> 4. copy fullchain.pem to /etc/pki/managerssl.crt
> 5. copy privkey.pem to /etc/pki/managerssl.key.
> 6. Then reboot (to restore original iptables config)

You my friend, saved my life. This is what I was looking for hours.
